src/main/java/com/jagrosh/vortex/commands/moderation/SlowmodeCmd.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 John Grosh (jagrosh).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.vortex.commands.moderation;
+
+import com.jagrosh.jdautilities.command.CommandEvent;
+import com.jagrosh.vortex.Vortex;
+import com.jagrosh.vortex.commands.ModCommand;
+import com.jagrosh.vortex.utils.FormatUtil;
+import com.jagrosh.vortex.utils.OtherUtil;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.TextChannel;
+
+/**
+ * @author Michaili K (mysteriouscursor+git@protonmail.com)
+ */
+public class SlowmodeCmd extends ModCommand
+{
+    private final static int MAX_SLOWMODE = 21600;
+
+    public SlowmodeCmd(Vortex vortex)
+    {
+        super(vortex, Permission.MESSAGE_MANAGE);
+        this.name = "slowmode";
+        this.arguments = "[time or OFF]";
+        this.help = "enables or disables slowmode";
+        this.botPermissions = new Permission[]{Permission.MANAGE_CHANNEL};
+        this.guildOnly = true;
+    }
+
+    @Override
+    protected void execute(CommandEvent event)
+    {
+        String args = event.getArgs();
+        TextChannel channel = event.getTextChannel();
+        int time;
+
+        // Event args is empty
+        if (args.isEmpty())
+        {
+            if (channel.getSlowmode() > 0)
+                time = 0;
+            else
+                time = 5;
+        }
+
+        // Event args isn't empty
+        else
+        {
+            // Because parseTime returns 0 if the input is invalid, we'll just check if the argument is not "0",
+            // then parse it & check again if it's 0. If it's 0, then it's likely invalid.
+            if (args.equalsIgnoreCase("off") || args.equals("0"))
+                time = 0;
+            else
+            {
+                // Because parseTime will return 0 when you don't give a time unit (for example, if you give "10" as
+                // argument, you expect for slowmode to be set to 10 seconds), we first try to simply parse the argument
+                // and if that fails, use parseTime
```

Comment:
> **parseTime should parse `10` as 10 seconds**, the behavior **requiring specific time units** originates in ArgsUtil, which runs a regex over the time input before attempting a parse
I'm a bit confused because this info seems conflicting to me. Do you expect for parseTime to return `10` when you give it 10?
Because parseTime doesn't do that, if I pass it `10` it returns 0. This is because, in the loop, it tries to retrieve the next "vals," which would be the time unit. But with `10`, there's no unit after that & causes an out of bounds exception:
![image](https://user-images.githubusercontent.com/39029839/97325639-7e0cb600-1873-11eb-9088-5a6e4c19e4ed.png)

Reply:
Ah, you're right. That is a bug in the parseTime method and should be fixed as well.